### PR TITLE
Exposed libcouchbase environment variables through build.rs mechanism.

### PR DESCRIPTION
**Pull request by tlebovic**
I have created this PR based on the discussion in #45.

### CODE REVIEW

**Comment by daschl** [comment]
@tlebovic this looks great, thanks! One idea that we can talk through briefly:

What about if instead of enumerating all lcb variables we iterate through all env variables and only pass down that have a common prefix like `LCB_`? The upside would be that we do not need to keep them in sync and users can pass down whatever they need, the downside is since some of them already have `LCB_` in the name you would have to end up passing something like `LCB_LCB_USE_ASAN` - or we find a different prefix? 

In any case I am okay with the change as is, I just wanted to propose a different way which might save us maintenance to keep them in-sync in the future.

**Comment by tlebovic** [comment]
I like the idea, just not a big fan of the possible double "LCB_" on some environment variables. Maybe the prefix should be "CBRS_"? The only other downside I can see with this approach is if you ever want to restrict users from setting specific build settings.

**Comment by daschl** [comment]
@tlebovic we could just use "CB_" ? .. I think if we ever want to restrict users we can blacklist them?

using "CB_" we can also open the door to add our own rust flags in the future, so it blends with the lcb ones.

**Comment by tlebovic** [comment]
@daschl that sounds good to me. I will submit changes probably tomorrow evening based on this discussion. Perhaps we will not implement the blacklisting right now, but it shouldn't be hard to add in the future if we need it.

File: couchbase-sys/build.rs [hunk 1]
```diff
@@ -3,6 +3,18 @@
 use std::env;
 use std::path::PathBuf;
 
+static ENV_FLAG_PREFIX: &'static str = "CB_";
+
+// Simple Environment Vairable filter
+fn environment_variable_filter(env_flag:&str) -> bool {
+    if env_flag.len() > ENV_FLAG_PREFIX.len() &&
```
Comment (daschl) [inline couchbase-sys/build.rs]: can this be simplified into just `env_flag.len() > ENV_FLAG_PREFIX.len() &&
       &env_flag[0..ENV_FLAG_PREFIX.len()] == ENV_FLAG_PREFIX` ? (avoid the if and return true false which is I think implicit in the logical and)

**Comment by daschl** [comment]
@tlebovic btw, have you signed the couchbase CLA? if not, can you please take a minute and follow the initial steps under https://github.com/couchbase/couchbase-java-client/blob/master/CONTRIBUTING.md#contributing ? Should not take long and once this is done you do not need to do it again.

Thanks much again for your great input and contributions!

File: couchbase-sys/build.rs (hunk 1 above)
Comment (tlebovic) [reply couchbase-sys/build.rs]: Yeah, I can make the change. The only reason I had it in the if/else block was to allow for whitelisting and subsequent filtering. But can remove it for now.

**Comment by daschl** [comment]
@tlebovic heyo let me know if you are stuck on things here to help you out

**Comment by tlebovic** [comment]
Sorry, got pulled off and didn't get the chance to circle back sooner. I will also complete the CLA in the next couple of days.

**Comment by sadolit** [comment]
Hi, are there any plans to continue with the PR?

**Comment by daschl** [comment]
Yes, I was waiting if @tlebovic was
Going to sign the CLA - once that is done we can merge it!

**Comment by daschl** [comment]
Since this is a rather contained change, @tlebovic for the interest of progress, if I don't get a response I'll just close this one out and get a similar change in under my name. I'll mention you in the commit if that's alright.

**Comment by daschl** [comment]
4da22840617eb8806a96c35e5576026fae170280 , thanks @tlebovic !

**Pull request closed without merging**